topic/service.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package topic
+
+import (
+	"github.com/m3db/m3cluster/kv"
+)
+
+type service struct {
+	store kv.Store
+}
+
+// NewService creates a topic service.
+func NewService(sOpts ServiceOptions) (Service, error) {
+	store, err := sOpts.ConfigService().Store(sOpts.KVOverrideOptions())
+	if err != nil {
+		return nil, err
+	}
+	return &service{
+		store: store,
+	}, nil
+}
+
+func (s *service) Get(name string) (Topic, int, error) {
```

Comment:
Comments on public functions? I would assume they are similar to those defined in the interface in the `types.go` file, but it would be nice to have those here too.

Reply:
Usually we don't repeat those comments if they are not changed from the interface.